### PR TITLE
fix: Fix coordinates and org. unit in event enrollments [DHIS2-10228]

### PR DESCRIPTION
**Pull request by maikelarabori**
This fix also includes DHIS2-10066.
With this fix, the event reports will bring results when DE of type Coordinate and OrgUnit are added as part of the columns.

**This is just a "forward-porting" from 2.33. It was already reviewed and merged in that version.**

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java [hunk 1]
```diff
@@ -342,6 +357,119 @@ protected String getSelectClause( EventQueryParams params )
         return "select " + StringUtils.join( selectCols, "," ) + " ";
     }
 
+    /**
+     * Returns an encoded column name respecting the geometry/coordinate format.
+     * The given QueryItem must be of type COORDINATE.
+     *
+     * @param item the {@link QueryItem}
+     * @return the column selector or EMPTY if the item valueType is not
+     *         COORDINATE. ie.: ( select '[' ||
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java:366]: Maybe shorten all this in the return Javadoc, a bit too much to include the entire SQL query.

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java [hunk 2]
```diff
@@ -550,6 +539,57 @@ else if ( item.hasOptionSet() && (option = item.getOptionSet().getOptionByCode(
...
+    protected String getCoordinateColumn( final QueryItem item, final String suffix )
+    {
+        final String colName = quote( item.getItemId() + suffix );
+
+        String stCentroidFunction = "";
+
+        if ( ValueType.ORGANISATION_UNIT == item.getValueType() )
+        {
+            stCentroidFunction = "ST_Centroid";
+
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java:574]: Remove blank line.

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java [hunk 3]
```diff
@@ -342,6 +357,119 @@ protected String getSelectClause( EventQueryParams params )
...
+
+            String stCentroidFunction = "";
+
+            if ( ValueType.ORGANISATION_UNIT == item.getValueType() )
+            {
+                stCentroidFunction = "ST_Centroid";
+
+            }
+
+            return "(select " +
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java:421]: Is it possible to format this in a more readable way? Or is the auto-formatter deciding it?

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java [hunk 4]
```diff
@@ -342,6 +357,119 @@ protected String getSelectClause( EventQueryParams params )
...
+    }
+
+    /**
+     * Creates a column "selector" for the given item name. The suffix will be
+     * appended as part of the item name. The column selection is based on
+     * events analytics tables.
+     *
+     * @param item the {@link QueryItem}
+     * @param suffix is currently ignored. Not currently used for enrollments
+     * @return 1) when there is a program stage: returns the column select
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java:438]: Reduce return Javadoc.

Comment (maikelarabori) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java]: Done

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java (hunk 3 above)
Comment (maikelarabori) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java]: Done

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java (hunk 2 above)
Comment (maikelarabori) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java]: Done

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java (hunk 1 above)
Comment (maikelarabori) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/JdbcEnrollmentAnalyticsManager.java]: Done

**Review by larshelge — APPROVED** [review]

**Pull request merged**